token-metadata/program/src/state.rs

```diff
@@ -0,0 +1,48 @@
+use {
+    borsh::{BorshDeserialize, BorshSerialize},
+    solana_program::pubkey::Pubkey,
+};
+/// prefix used for PDAs to avoid certain collision attacks (https://en.wikipedia.org/wiki/Collision_attack#Chosen-prefix_collision_attack)
+pub const PREFIX: &str = "metadata";
```

Comment:
:star_struck:  Nice forethought!  Hash domaining and collision resistance aren't considered nearly enough in this space